### PR TITLE
chore: enable more ruff rules, fix

### PR DESCRIPTION
**Pull request by akx**
Some of these are manual fixes, some are automated. 

Would recommend manual review! Each commit should be reviewable in isolation, and there should be no back-and-forth between them.

### CODE REVIEW

**Comment by akx** [comment]
Might be a good idea to get #340 merged first so Ruff linting is enforced in CI.

File: backend/access/cbac.py [hunk 1]
```diff
@@ -135,7 +136,7 @@ def is_graphql_allowed_for_model(
 def graphql_check_access(instance, info, field: str):
     user = info.context.user
     operation = "query"
-    extra = dict(slug=instance.slug) if hasattr(instance, "slug") else {}
+    extra = {"slug": instance.slug} if hasattr(instance, "slug") else {}
```
Comment (japsu) [inline backend/access/cbac.py:139]: This is a personal preference and I will keep writing `dict(foo=5)` :) But no need to remove this from the PR.

Comment (akx) [reply backend/access/cbac.py:139]: Yeah, this got changed automagically somewhere else as well, and I reverted it then too. Will revert this too while rebasing, no biggie!

Comment (akx) [reply backend/access/cbac.py]: Also, the personal preference is noted by
```toml
ignore = [
  # ...
  "C408",  # unnecessary dict call (these are likely stylistic)
]
```
😌

**Comment by japsu** [comment]
image

404 on `/graphql` is a bit rough, no idea what's causing it but obv can't merge as-is @akx

**Review by japsu — CHANGES_REQUESTED** [review]
need to fix 404 on `/graphql`

**Comment by akx** [comment]
> 404 on `/graphql` is a bit rough, no idea what's causing it but obv can't merge as-is

weh, it was getting mounted at a wrong path due to a brainfart:

Screenshot 2024-01-16 at 19 10 57

Rebased, fixed that, added a new commit that tests that the GraphQL API is mounted at the well-known path `/graphql`...

**Review by japsu — APPROVED** [review]

**Pull request merged**